src/main/webapp/src/app/hidden-gem.service.ts

```diff
@@ -11,16 +12,15 @@ import { catchError } from 'rxjs/operators';
 export class HiddenGemService {
   constructor(private httpClient: HttpClient) { }
 
-  // TODO: Remove function when not needed anymore (and update spec.ts file). Function kept for debugging purposes.
-  getAllHiddenGems() : Observable<HiddenGem[]> {
-    return this.httpClient
-      .get<HiddenGem[]>('/hiddengems')
-      .pipe(catchError(this.handleError))
-  }
+  getRankedHiddenGems(location: Location): Observable<HiddenGem[]> {
+    const lat = location.lat;
+    const lng = location.lng;
+    const params = new HttpParams()
+      .set('lat', lat.toString())
+      .set('lng', lng.toString());
 
-  getRankedHiddenGems() : Observable<HiddenGem[]> {
     return this.httpClient
-      .get<HiddenGem[]>('/ranking')
+      .get<HiddenGem[]>('/ranking', {params})
       .pipe(catchError(this.handleError))
   }
 
```

Comment:
I cant seem to add a comment on the line. Can you add a TODO for the findHiddenGemRecommendation - so we remember to remove the function when we remove the servlet for recommendation. Also maybe add a TODO in the form component, that subscribes to the findHiddenGemRecommendation - it will probbaly need to be updated..